pkg/cmd/cli/datamover/backup.go

```diff
@@ -73,20 +99,187 @@ func NewBackupCommand(f client.Factory) *cobra.Command {
 	return command
 }
 
+const (
+	// defaultCredentialsDirectory is the path on disk where credential
+	// files will be written to
+	defaultCredentialsDirectory = "/tmp/credentials"
+)
+
 type dataMoverBackup struct {
-	logger logrus.FieldLogger
-	config dataMoverBackupConfig
+	logger      logrus.FieldLogger
+	ctx         context.Context
+	cancelFunc  context.CancelFunc
+	client      ctlclient.Client
+	cache       ctlcache.Cache
+	namespace   string
+	nodeName    string
+	config      dataMoverBackupConfig
+	kubeClient  kubernetes.Interface
+	dataPathMgr *datapath.Manager
 }
 
-func newdataMoverBackup(logger logrus.FieldLogger, config dataMoverBackupConfig) *dataMoverBackup {
+func newdataMoverBackup(logger logrus.FieldLogger, factory client.Factory, config dataMoverBackupConfig) (*dataMoverBackup, error) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	clientConfig, err := factory.ClientConfig()
+	if err != nil {
+		cancelFunc()
+		return nil, errors.Wrap(err, "error to create client config")
+	}
+
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
```

Comment:
It seems this line is not relevant to the context.

Reply:
Let me check this separately since this code is same with the node-agent server.
If there is any problem, I will modify it for both here and the node-agent server in a separate PR.

Reply:
After a preliminary check, this is basically OK.` zap.New(zap.UseDevMode(true))` also initialize a logger to stderr which is the sam with `logger logrus.FieldLogger` and the default log level is `Info`.
Merely, `logger logrus.FieldLogger` is with more info about the log level and format, but ` zap.New(zap.UseDevMode(true))` may not comply with them. We probably seldomly need the logs from controller-runtime to be in `debug` level.

The same code in the node-agent server was introduced in this PR https://github.com/vmware-tanzu/velero/pull/2561. Since it is an old one, we don't know any other special reason for not using the provided logger.

Therefore, I suggest we keep it as is until we see problems, then we come back to change all the places the same as here.